fix #278023: text style usability issues

MarcSabatella opened the pull request:
See https://musescore.org/en/node/278023.  Now that Werner has added the "missing" style values (looks like he got them all except perhaps spatiumdependent and offsettype, which maybe we don't need), I have gone in and cleaned up the style editor and inspector to be more consistent.  See my comments in issue tracker regarding what I did and haven't done.  To me this is usable as is, more would be gravy and someone else can maybe deal with it if desired.

anatoly-os commented:
I reviewed the changes a bit. I don't like `for` loop on enum values. It is error-prone approach because enum values are not guaranteed at all to be defined one-by-one. Better approach is to create safe data structure (like vector or map) and iterate over it using `for (auto val : container)`.

Regarding huge list in the inspector, I really cannot stop using "Search" in palettes. It is amazing thing helping to find necessary element in palettes. Styles combobox in Inspector could be replaced with additional window listing all styles. This small window may have search textfield to navigate over all styles. To combine both UX elements, we can list "most used" text styles in combobox and offer "Choose more..." button to open a list of all styles to choose.

MarcSabatella commented:
Updated as per comment above - I now have functions allTextStyles() and primaryTextStyles() defined in style.cpp that the style dialog and inspector iterate over.  I still use a sequential "for" loop in allTextStyles() because at that point in the code, I am not actually iterating through an enum but through the existing textStyles array, which is by definition sequential.

I also went ahead and added a "Style" control to the inspector for tempo text, dynamics, and lyrics, and also made the order of the controls more consistent in a couple of places.  I also restricted the list of available styles for fingering (back to how it was) because I know the placement algorithm depends on this.

We could certainly still add a "More" control to allows the inspector to access all styles.  Not totally sure it's worth the effort.  The list as I now have is a bit bigger than what was there originally and I did try to think logically about what should be included.

MarcSabatella commented:
Some more cleanup, plus adding the ability to set placement & style for jumps & markers (repeat text).  However, this last commit doesn't quite work yet, bad results on save/load.

MarcSabatella commented:
Reverted the change to set style & placement for repeats, there are some underlying issues to address in the handling of property defaults.  We can revisit this later if desired.

MarcSabatella commented:
I will look at these conflicts and resolve them, probably tomorrow, doesn't look too bad.

MarcSabatella commented:
Rebased with the recent Inspector changes.  This plus #4207 gets the Inspector close to where it needs to be.  Still need to figure out the strategy for the reset / set as style controls, but I can submit a PR later for my proposed solution.
